packages/runner-shared/src/event-manager.js

```diff
@@ -526,6 +530,15 @@ export const eventManager = {
       localBus.emit('expect:domain', domain)
     })
 
+    Cypress.multiDomainCommunicator.on('viewport:changed', (viewport, domain) => {
+      const callback = () => {
+        Cypress.multiDomainCommunicator.toSpecBridge(domain, 'viewport:changed:end')
+      }
+
+      Cypress.multiDomainCommunicator.emit('sync:viewport', viewport)
```

Comment:
I feel conflicted about the use of `multiDomainCommunicator` to send messages within a domain. I like the expectation that when you get a message from `multiDomainCommunicator` you can assume it has come from another domain and that the 3rd parameter is the domain the event has originated from, this event `sync:viewport` and the `expect:domain` event break that expectation.

Reply:
My thought of using the `multiDomainCommunicator` was that it would be easy to identify this message is used for multi-domain and not needed for general Cypress. Though I do agree not having the `domain` does break that expectation. I'll add a TODO comment here so we can discuss further and not hold up this PR.